docs: update to include Supabase S3 File Provider (#8715)
medusajs/medusa: www/apps/resources/app/architectural-modules/file/s3/page.mdx

    <TabsTrigger value="aws">AWS S3</TabsTrigger>
    <TabsTrigger value="minio">MinIO</TabsTrigger>
    <TabsTrigger value="spaces">DigitalOcean Spaces</TabsTrigger>

>     <TabsTrigger value="suapbase">Supabase S3 Storage</TabsTrigger>

Reviewer: ```suggestion
    <TabsTrigger value="supabase">Supabase S3 Storage</TabsTrigger>
```

    - Create [DigitalOcean Spaces bucket](https://docs.digitalocean.com/products/spaces/how-to/create/).
    - Create [DigitalOcean Spaces access and secret access keys](https://docs.digitalocean.com/products/spaces/how-to/manage-access/#access-keys).

    </TabsContent>

> <TabsContent value="supabase">

Reviewer: ```suggestion
    <TabsContent value="supabase">
```

>     - [Supabase account](https://supabase.com/dashboard/sign-in?returnTo=%2Fprojects).

Reviewer: ```suggestion
    - [Supabase account](https://supabase.com/dashboard/sign-in) with a project.
```

>     - Create [New Project]

Reviewer: ```suggestion
```

    - Create [New Public S3 bucket](https://supabase.com/docs/guides/storage/buckets/creating-buckets?queryGroups=language&language=dashboard).
    - Create [New S3 Access Keys](https://supabase.com/docs/guides/storage/s3/authentication?queryGroups=language&language=javascript#s3-access-keys).
    - Create [Storage Policy](https://supabase.com/docs/guides/storage/security/access-control).

>       1. On your bucket's dashboard, under configuration, click polocies.

Reviewer: ```suggestion
      1. On your bucket's dashboard, click on Policies under the Configuration sidebar.
```

>       2. Click on New Policy under Other policies under storage.objects.

Reviewer: ```suggestion
      2. Click on New Policy under "Other policies under storage.objects".
```

>       3. Click get started quickly and enable insert access for authenticated users only.

Reviewer: ```suggestion
      3. Click get started quickly, choose "insert access for authenticated users only", and click the "Use this template" button.
```

>       4. Click ALL instead of INSERT, then review and save policy.

Reviewer: ```suggestion
      4. Click ALL for "Allowed Operations", click the "Review" button, then the "Save policy" button.
```

      5. Click settings under configuration.
      6. Click New access key.
      7. Name new access key, and save id and secret key. 
      8. S3_FILE_URL= ``https://{uniqueID}/supabase.co/storage/v1/object/public``

>       9. Copy settings for ``S3_ENDPOINT``, ``S3_REGION``, ``S3_BUCKET``, ``S3_ACCESS_KEY``, ``S3_SECRET_KEY``

Reviewer: ```suggestion
```

No need for these as you already added the link to the S3 access key guide above.

      - For AWS S3, the endpoint is of the format `https://{bucket}.s3.{region}.amazonaws.com`
      - For MinIO, it's the URL to the MinIO server with the bucket's name. For example, `https://{server-domain}/{bucket}`.
      - For DigitalOcean Spaces, it's either the Origin Endpoint or the CDN endpoint of your Spaces Object Storage bucket.

>       - for Supabase, it's `https://{uniqueID}.supabase.co/storage/v1/object/public/{bucket}`.

Reviewer: ```suggestion
      - for Supabase, it's `https://{uniqueID}.supabase.co/storage/v1/object/public/{bucket}`. You can retrieve the `uniqueID` from [Storage Settings](https://supabase.com/docs/guides/storage/s3/authentication?queryGroups=language&language=javascript#s3-access-keys) page in the Endpoint URL.
```

      - For AWS S3, the endpoint is of the format `https://s3.{region}.amazonaws.com`
      - For MinIO, it's the URL to the MinIO server.
      - For DigitalOcean Spaces, it's the Spaces Origin Endpoint.

>       - For Supabase, it's the Endpoint url

Reviewer: ```suggestion
      - For Supabase, it's the Endpoint URL in the [Storage Settings](https://supabase.com/docs/guides/storage/s3/authentication?queryGroups=language&language=javascript#s3-access-keys).
```

  </Table.Body>
</Table>

## Additional Configuration for MinIO and Supabase

If you're using MinIO or Supabase, set `forcePathStyle` to `true` in the `additional_client_config` object.

For example:

```js title="medusa-config.js"
module.exports = defineConfig({
  // ...
  modules: {
    [Modules.FILE]: {
      resolve: "@medusajs/file",
      options: {
        providers: [
          {
            resolve: "@medusajs/file-s3",
            id: "s3",
            options: {
              // ...
              additional_client_config: {
                forcePathStyle: true,
              },
            },
          },
        ],
      },
    },
  }
})

> ```

Reviewer: ~~~suggestion
```
---

~~~
